contracts/native/FluxMegaOracle.sol

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.6;
+
+import "../utils/Governable.sol";
+import "../interfaces/native/IFluxPriceFeed.sol";
+import "../interfaces/native/IFluxMegaOracle.sol";
+
+
+/**
+ * @title FluxMegaOracle
+ * @author solace.fi
+ * @notice An oracle that consumes data from [Flux](https://fluxprotocol.org) and returns it in a useable format.
+ *
+ * [Governance](/docs/protocol/governance) can add or remove price feeds via [`addPriceFeeds()`](#addpricefeeds) and [`removePriceFeeds()`](#removepricefeeds). Users can view price feeds via [`priceFeedForToken(address token)`](#pricefeedfortoken). Users can use the price feeds via [`valueOfTokens()`](#valueoftokens).
+ */
+contract FluxMegaOracle is IFluxMegaOracle, Governable {
+
+    // token => data
+    mapping(address => PriceFeedData) internal _priceFeeds;
+    // index => token
+    mapping(uint256 => address) internal _indexToToken;
+    // token => index+1
+    mapping(address => uint256) internal _tokenToIndex;
+    uint256 internal _tokensLength;
+
```

Comment:
Is it simpler to use an EnumerableSet.AddressSet in place of `_indexToToken `, `_tokenToIndex `, and `_tokensLength`. It would also simplify the implementations for `addPriceFeeds()` and `removePriceFeeds()` imo.

Reply:
you're probably right I should've started with that. but now that this is built its more work to swap it in